index.js

```diff
@@ -49,42 +49,27 @@ if (!config) {
   throw new Error('you did\'t define any krabby reports. great job.');
 }
 
-tests = config.tests.map(function(test) {
-  var testName;
-  var config = {};
-
-  if (typeof test === 'string') {
-    testName = test;
-  } else {
-    testName = test.name;
-
-    config = test;
-  }
-
-  var Test = require(Path.join(__dirname, 'lib/tests', testName));
-
-  test = new Test(config);
-
-  return test.test.bind(test);
-});
-
-reports = config.reports.map(function(report) {
-  var reportName;
-  var config = {};
-
-  if (typeof report === 'string') {
-    reportName = report;
-  } else {
-    reportName = report.name;
-    config = report;
-  }
-
-  report = require(Path.join(__dirname, 'lib/reports', reportName));
-  report = report(config);
-
-  return report;
-});
+var findPlugins = function(plugins, path, rootMethod) {
+  return plugins.map(function(plugin) {
+    var pluginName;
+    var config = {};
+
+    if (typeof plugin === 'string') {
+      pluginName = plugin;
+    } else {
+      pluginName = plugin.name;
+      config = plugin;
+    }
+
+    var Plugin = require(Path.join(__dirname, path, pluginName));
+    plugin = new Plugin(config);
+
+    return plugin[rootMethod].bind(plugin);
```

Comment:
We could make this even more simple by figuring out what a good method name should be for both tests AND reports